src/map.rs

```diff
@@ -1096,8 +1096,12 @@ where
         K: Borrow<Q>,
         Q: Hash + Eq,
     {
-        let hash = make_hash::<K, Q, S>(&self.hash_builder, k);
-        self.table.get(hash, equivalent_key(k))
+        if self.table.is_empty() {
+            None
+        } else {
+            let hash = make_hash::<K, Q, S>(&self.hash_builder, k);
+            self.table.get(hash, equivalent_key(k))
+        }
```

Comment:
it may make sense, if the map contains a single element and the table is small, to skip hashing and compare directly the key with the key of the single entry in the table